hist/histv7/inc/ROOT/RHistImpl.hxx

```diff
@@ -320,29 +320,29 @@ enum class EBinCoord {
    kBinTo      ///< Get the bin high edge
 };
 
-template <int I, class COORD, class AXES>
+template <int I, class HISTIMPL, class COORD, class AXES>
 struct RFillBinCoord;
 
 // Break recursion.
-template <class COORD, class AXES>
-struct RFillBinCoord<-1, COORD, AXES> {
+template <class HISTIMPL, class COORD, class AXES>
+struct RFillBinCoord<-1, HISTIMPL, COORD, AXES> {
    void operator()(COORD & /*coord*/, const AXES & /*axes*/, EBinCoord /*kind*/, int /*binidx*/) const {}
 };
 
 /** Fill `coord` with low bin edge or center or high bin edge of all axes.
  */
-template <int I, class COORD, class AXES>
+template <int I, class HISTIMPL, class COORD, class AXES>
 struct RFillBinCoord {
    void operator()(COORD &coord, const AXES &axes, EBinCoord kind, int binidx) const
    {
-      int axisbin = binidx % std::get<I>(axes).GetNBins();
-      size_t coordidx = std::tuple_size<AXES>::value - I - 1;
+      constexpr const int thisAxis = HISTIMPL::GetNDim() - I - 1;
```

Comment:
Could we pass `int NDIM` instead of the complete `HISTIMPL`? Do we even need that, or can we just pass `thisAxis` as template parameter? (Sorry not enough coffee to figure that one out myself.)

Reply:
NDIM is totally possible. I don't think we can get by with just "thisAxis", because in order to loop from 0 to NDIM-1 we need to know what NDIM is.

Reply:
(this is unlike when looping in the reverse direction from NDIM-1 to 0, where we can just decrement until hitting -1)

Reply:
Fixed in latest commit.